app/controllers/application_controller.rb

```diff
@@ -77,11 +77,17 @@ def ensure_visit_integrity
     end
   end
 
-  def pad_execution_time(execution_time)
+  def pad_execution_time(execution_time_limit, max_pad_multiple=5)
     start = Time.zone.now
     result = yield
     stop = Time.zone.now
-    sleep execution_time - (stop - start)
+    pad_time = execution_time_limit - (stop - start)
+    if pad_time < 0
+      pad_time = rand(max_pad_multiple) * execution_time_limit
+      Rails.logger.info "Execution padding insufficient"
```

Comment:
Hm. That's annoying; it'd be nice to have the stack trace—or at least the caller—so we can identify which calls are overrunning. Any ideas?